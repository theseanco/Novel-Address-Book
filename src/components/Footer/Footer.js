/*

Footer component containing my name and the current year

*/

import React from 'react';

const Header = () => {

  // Get current year
  const date = new Date().getFullYear();

  return (
    <footer data-testid="footer">
      <h5>Sean Cotterill, {date}</h5>
    </footer>
  )
}

export default Header;