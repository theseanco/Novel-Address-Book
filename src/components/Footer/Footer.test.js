import React from 'react';
import { render, getByText } from '@testing-library/react';
import Footer from './Footer';

describe('footer tests', () => {
  // Mock date function
  global.Date.getFullYear = jest.fn(() => 2020);

  // Using regex to match parts of text rather than complete text
  it('should return my name and the date', () => {
    const { container } = render(<Footer />);
    expect(getByText(container, /Sean Cotterill/)).toBeInTheDocument()
    expect(getByText(container, /2020/)).toBeInTheDocument()
  })

  // Snapshot test for the footer component as it does not have stateful UI logic
  it('should match snapshot', () => {
    const { container } = render(<Footer />);
    expect(container.firstChild).toMatchSnapshot();
  })
});