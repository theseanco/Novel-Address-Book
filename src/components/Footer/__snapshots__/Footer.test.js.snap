// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer tests should match snapshot 1`] = `
<footer
  data-testid="footer"
>
  <h4>
    Sean Cotterill, 
    2020
  </h4>
</footer>
`;
